### PR TITLE
Feature/multiple options

### LINKED ISSUE #156: Support multiple seeds

### PR DESCRIPTION
**Pull request by area363**
This PR migrates from `CommandLineParser` to `Cocona.Lite` to take multiple `seed` and `ice-server` arguments.

Example args:
Screen Shot 2021-02-23 at 11 36 23 PM

### CODE REVIEW

File: Libplanet.Explorer.Executable/Program.cs [hunk 1]
```diff
@@ -30,11 +29,135 @@ namespace Libplanet.Explorer.Executable
...
+            mysqlPassword,
+            mysqlDatabase,
+            maxTransactionsPerBlock,
+            maxBlockBytes,
+            maxGenesisBytes,
+            seedStrings,
+            iceServerUrl,
+            storePath,
+            storeType,
+            genesisBlockPath);
```
Comment (moreal) [inline Libplanet.Explorer.Executable/Program.cs]: It seems to need one more indent. How do you think?

File: Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <NoWarn>NU1701,SA1118</NoWarn>
+    <NoWarn>NU1701,SA1118,SA1201,MEN003</NoWarn>
```
Comment (moreal) [inline Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj]: Is there reason to turn off `SA1201`? In my thought, you should follow the rule if you can.

File: Libplanet.Explorer.Executable/Program.cs [hunk 3]
```diff
@@ -174,8 +297,7 @@ private static IRichStore LoadStore(Options options)
                 default:
                     // FIXME: give available store type as argument hint without code duplication.
                     var availableStoreTypes = new[] { "rocksdb", "default" };
-                    string longOptionName = options.GetType().GetProperty(nameof(options.StoreType))
-                        .GetCustomAttribute<OptionAttribute>().LongName;
+                    string longOptionName = "store-type";
```
Comment (moreal) [inline Libplanet.Explorer.Executable/Program.cs]: If you use the name as constant, without reflection. It seems better to use with `const` keyword.

File: Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj (hunk 2 above)
Comment (ipdae) [inline Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj]: 이 룰추가는 의도된건가요?

File: Libplanet.Explorer.Executable/Program.cs [hunk 4]
```diff
@@ -30,11 +29,135 @@ namespace Libplanet.Explorer.Executable
...
+                Description = "The number of maximum bytes size of the genesis block.")]
+            int maxGenesisBytes = 1024 * 1024,
+            [Option(
+                "seed",
+                new[] { 's' },
+                Description = @"Seed nodes to join to the network as a node. The format of each
+seed is a comma-separated triple of a peer's hexadecimal public key, host, and port number.
+E.g., `02ed49dbe0f2c34d9dff8335d6dd9097f7a3ef17dfb5f048382eebc7f451a50aa1,example.com,31234'.
+If omitted (default) explorer only the local blockchain store.")]
+            string[] seedStrings = null,
```
Comment (moreal) [inline Libplanet.Explorer.Executable/Program.cs:136]: When I test this in my local, `seedStrings` is optional and it occurs `System.ArgumentNullException: Value cannot be null.` at the setter of `Options.SeedStrings`. Could you handle this?

File: Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj (hunk 2 above)
Comment (area363) [reply Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj]: Screen Shot 2021-02-24 at 11 02 43 AM

`MEN003`은 [Run 메소드](https://github.com/planetarium/libplanet-explorer/pull/160/files#diff-4592f143a31d8fc9c98f043229eeb7041b611025ba1256630ca1772f722cfff9R40)가 120줄 이하여야 한다고 나오는데 지금 240줄 정도여서 더 줄일 방도가 안보여서 추가했습니다.

`SA1201` 경우에는 [constructor](https://github.com/planetarium/libplanet-explorer/pull/160/files#diff-dd6f52bbdaedce000faaaf36afae83df9b3576689c9e5501e6f590e5e63c34f0R127-R172)가 [property](https://github.com/planetarium/libplanet-explorer/pull/160/files#diff-dd6f52bbdaedce000faaaf36afae83df9b3576689c9e5501e6f590e5e63c34f0L16-R125) 위로 올라와야한다고 하는데 그렇게 하는게 맞을까요? @moreal

File: Libplanet.Explorer.Executable/Program.cs (hunk 4 above)
Comment (area363) [reply Libplanet.Explorer.Executable/Program.cs:136]: Thanks, I'll look into this part!

File: Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj (hunk 2 above)
Comment (moreal) [reply Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj]: > `SA1201` 경우에는 [constructor](https://github.com/planetarium/libplanet-explorer/pull/160/files#diff-dd6f52bbdaedce000faaaf36afae83df9b3576689c9e5501e6f590e5e63c34f0R127-R172)가 [property](https://github.com/planetarium/libplanet-explorer/pull/160/files#diff-dd6f52bbdaedce000faaaf36afae83df9b3576689c9e5501e6f590e5e63c34f0L16-R125) 위로 올라와야한다고 하는데 그렇게 하는게 맞을까요? @moreal

네, 저는 `SA1201`를 따르는 게 맞다고 생각합니다.

File: Libplanet.Explorer.Executable/Program.cs [hunk 5]
```diff
@@ -30,11 +29,135 @@ namespace Libplanet.Explorer.Executable
...
+        [Command(Description = "Run libplanet-explorer with options.")]
+        public async Task Run(
+            [Option(
+                "seed",
+                new[] { 's' },
+                Description = @"Seed nodes to join to the network as a node. The format of each
+seed is a comma-separated triple of a peer's hexadecimal public key, host, and port number.
+E.g., `02ed49dbe0f2c34d9dff8335d6dd9097f7a3ef17dfb5f048382eebc7f451a50aa1,example.com,31234'.
+If omitted (default) explorer only the local blockchain store.")]
+            string[] seedStrings,
```
Comment (moreal) [inline Libplanet.Explorer.Executable/Program.cs]: Is it okay to be _required_ option? In my memory, libplanet-explorer could be run in local (i.e. offline mode), not in network, to explore local storage.

File: Libplanet.Explorer.Executable/Program.cs [hunk 6]
```diff
@@ -30,11 +29,135 @@ namespace Libplanet.Explorer.Executable
...
+                "store-type",
+                new[] { 'T' },
+                Description = @"The type of the blockchain store. If omitted (default)
+in DefaultStore is used.")]
+            string storeType = null,
+            [Option(
+                "genesis-block",
+                new[] { 'G' },
+                Description = "The path of the genesis block. It should be absolute or http url.")]
+            string genesisBlockPath = null
```
Comment (moreal) [inline Libplanet.Explorer.Executable/Program.cs]: IMO these options seem required.

File: Libplanet.Explorer.Executable/Program.cs (hunk 5 above)
Comment (area363) [reply Libplanet.Explorer.Executable/Program.cs]: Ah I understand. I'll fix this accordingly.

**Review by moreal — APPROVED** [review]

**Review by longfin — APPROVED** [review]

**Review by ipdae — APPROVED** [review]

**Review by longfin — APPROVED** [review]

File: Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj [hunk 7]
```diff
@@ -4,7 +4,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <NoWarn>NU1701,SA1118</NoWarn>
+    <NoWarn>NU1701,SA1118,MEN003</NoWarn>
```
Comment (dahlia) [inline Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj]: We should not turn off this warning.  Which method does violate this?  [We can suppress a single source file or lines of a file.](https://docs.microsoft.com/en-us/visualstudio/code-quality/in-source-suppression-overview)

Comment (area363) [reply Libplanet.Explorer.Executable/Libplanet.Explorer.Executable.csproj]: The [Run method](https://github.com/planetarium/libplanet-explorer/pull/160/files#diff-4592f143a31d8fc9c98f043229eeb7041b611025ba1256630ca1772f722cfff9R40) in `Libplanet.Explorer.Executable/Program.cs` is currently around 240 lines but `MEN003` enforces a 120-line limit. I'm wasn't sure how to reduce the lines that much so I added this part. But I'll look into suppressing that method individually like you just suggested. Thanks!

**Review by dahlia — APPROVED** [review]

**Pull request merged**